### PR TITLE
Dispersy test expansion to support community testing better

### PR DESCRIPTION
**Pull request by snorberhuis**
These changes allow communities to be better tested.

This is used to test the MultiChain community:  Tribler/tribler#1429.

### CODE REVIEW

**Comment by snorberhuis** [comment]
I will remove the WIP commit.

**Comment by whirm** [comment]
Ok, so that's what's happening: The delayed call gets canceled from the taskmanager, but the delayed call that calls the start method: 

`reactor.callLater(0, lc.start, DOWNLOAD_MM_PK_INTERVAL, now=True)` in community:346

Hasn't been called yet when `on_identity()` gets called, so the looping call gets started even after being canceled from the taskManager.

**Comment by snorberhuis** [comment]
So this is a separate issue with Dispersy that is a potential bug? 

If I should fix it here, what is the solution? Because there is probably a reason the `reactor.callLater` is used. Can the `callLater` also be removed when canceling the looping call?

**Comment by whirm** [comment]
TaskManager should be extended to create and deal with delayed looping calls so it knows it has to cancel the associated DelayedCall, I'll do it after I fix the Tribler bug I'm working on now. Should be done later today.

**Comment by whirm** [comment]
@snorberhuis: #457 should fix it.

**Comment by whirm** [comment]
@snorberhuis merged

**Comment by snorberhuis** [comment]
I tested with #457 and it does work. I reverted the old bugfix.

File: tests/dispersytestclass.py [hunk 1]
```diff
@@ -18,7 +25,7 @@
 # use logger.conf if it exists
 if os.path.exists("logger.conf"):
     # will raise an exception when logger.conf is malformed
-    logging.config.fileConfig("logger.conf")
+    logging.basicConfig(filename="logger.conf")
```
Comment (whirm) [inline tests/dispersytestclass.py:28]: @snorberhuis what issue where you having with this?

Also, better get used to split commits into different features, not a bunch of unrelated stuff.

File: tests/dispersytestclass.py [hunk 2]
```diff
@@ -63,24 +71,43 @@ def tearDown(self):
...
 
                 nodes.append(node)
             self._logger.debug("create_nodes, nodes created: %s", nodes)
             returnValue(nodes)
 
-        return blockingCallFromThread(reactor, _create_nodes, amount, store_identity, tunnel, communityclass, autoload_discovery)
+        return blockingCallFromThread(reactor, _create_nodes, amount, store_identity, tunnel, community_class,
+                                      autoload_discovery, memory_database)
+
+    def _create_node(self, dispersy, community_class, c_master_member):
```
Comment (whirm) [inline tests/dispersytestclass.py:112]: Are you using this somewhere else or it's a cleanup leftover?

Comment (snorberhuis) [inline tests/dispersytestclass.py:112]: I used it somewhere else.

I used it in my own tests to setup a DebugNode with a specific curve by overriding the method.

File: tests/dispersytestclass.py (hunk 1 above)
Comment (snorberhuis) [inline tests/dispersytestclass.py:28]: This code was normally not executed during tests for dispersy, because a logger.conf is not available by default in Dispersy.
But I started testing from Tribler where one is available by default. This resulted in an exception, because the logging.config is not imported correctly.

If config.fileConfig is used, then during test execution all debug statements are logged. While if basicConfig is used, the tests have clean logging statements.

Comment (snorberhuis) [inline tests/dispersytestclass.py:28]: Working for more than a week on this resulted in poor commit etiquette. Sorry!

File: tests/dispersytestclass.py (hunk 2 above)
Comment (whirm) [inline tests/dispersytestclass.py:112]: ok

**Comment by whirm** [comment]
Ready to merge then?
